File: Unit2-01/FlynnStack.ts
/**
 * This is the FlynnStack class.
 *
 * author Flynn Rundquist
 * version 1.0
 * since 2024-03-19
 */

class FlynnStack {
    strStack: string[]
    
    constructor() {
    private strStack: string[] = []
    constructor() {
    }

    pop(): string {
        return this.strStack.pop()
    }
    push(input: string): void {
        this.strStack.push(input)
    }

    show(): string {
        let values = ""
        for (let counter = 0; counter < this.strStack.length; counter++) {
                values += this.strStack[counter] + ", "
        }
        values = values.substring(0, values.length - 2)
        return values
    }
}

export default = FlynnStack

File: Unit2-01/FlynnStack.java
/**
 * The FlynnStack class.
 *
 * @author Flynn Rundquist
 * @version 1.0
 * @since 2024-03-22
 */

final class FlynnStack {
    /**
     * Private constructor prevents instantiation of this class.
     */
    private String[] stack;

    /**
     * Constructor.
     *
     * @param size The size of the stack
     */
<<<<<<< HEAD
    FlynnStack() {
=======
    FlynnStack(int size) {
>>>>>>> 946cd052f05dab6c5da18fd9b35e975f095d29cc
        stack = new String[];
    }

    /**
     * Method to push an item.
     *
     * @param item The item
     */
    public void push(String item) {
        for (int counter = 0; counter < stack.length; counter++) {
            if (stack[counter] == null) {
                stack[counter] = item;
                break;
            }
        }
    }

    /**
     * Method to show the stack.
     *
     * @return The stack
     */
    public String show() {
        String output = "";
        for (int counter = 0; counter < stack.length; counter++) {
            if (stack[counter] != null) {
                output += stack[counter] + ", ";
            }
        }
        if (!output.isEmpty()) {
            output = output.substring(0, output.length() - 2);
        }
        return output;
    }
}

